pkg/apis/v1alpha1/types.go

```diff
@@ -177,12 +177,39 @@ type MeshServiceList struct {
 	Items []MeshService `json:"items"`
 }
 
+func MergeSecret(a, b *corev1.Secret) *corev1.Secret {
```

Comment:
Ripped off from the `MergeService` logic below. Just need to determine what the final state of the `Secret` should be if we're modifying one that already exists.